src/Symfony/Component/Yaml/Inline.php

```diff
@@ -635,16 +646,17 @@ private static function evaluateScalar($scalar, $flags, $references = array())
      * @param string $value
      * @param int    &$i
      * @param int    $flags
+     * @param string $stopOn
      *
      * @return null|string
      */
-    private static function parseTag($value, &$i, $flags)
+    private static function parseTag($value, &$i, $flags, $stopOn = '')
     {
         if ('!' !== $value[$i]) {
             return;
         }
 
-        $tagLength = strcspn($value, " \t\n", $i + 1);
+        $tagLength = strcspn($value, " \t\n".$stopOn, $i + 1);
```

Comment:
In fact indicators must always be forbidden in tags handle (see http://www.yaml.org/spec/1.2/spec.html#ns-tag-char).
It should be done in 3.3 as a bug fix imo.